lib/net/ssh/service/forward.rb

```diff
@@ -91,7 +91,10 @@ def local(*args)
 
         channel.on_open_failed do |ch, code, description|
           channel.error { "could not establish direct channel: #{description} (#{code})" }
-          channel[:socket].close
+          session.stop_listening_to(ch[:socket])
+          channel[:socket].close unless channel[:socket].closed?
+          channel.do_close
+          session.channels.delete(ch.local_id)
```

Comment:
https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh/connection/session.rb#L561
we already delete channel, so this should not be neccesary